services/streammanager/googlesheets/googlesheetsmanager.go

```diff
@@ -161,46 +159,93 @@ func insertDataToSheet(sheetsClient *sheets.Service, spreadSheetId string, sprea
 		return fmt.Errorf("[GoogleSheets] error  :: Failed to initialize google-sheets client")
 	}
 
-	if isHeader {
-		_, err = sheetsClient.Spreadsheets.Values.Update(spreadSheetId, spreadSheetTab+"!A1", &vr).ValueInputOption("RAW").Do()
+	_, err = sheetsClient.Spreadsheets.Values.Update(spreadSheetId, spreadSheetTab+"!A1", &vr).ValueInputOption("RAW").Do()
+
+	return err
+}
+
+// Wrapper func to append row data list,
+// Returns error for failure cases of API calls otherwise returns nil
+func insertRowDataToSheet(sheetsClient *sheets.Service, spreadSheetId string, spreadSheetTab string, dataList [][]interface{}) error {
+	// Creating value range for inserting row into sheet
+	var vr sheets.ValueRange
+	vr.MajorDimension = "ROWS"
+	vr.Range = spreadSheetTab + "!A1"
+	vr.Values = dataList
+	var err error
 
-	} else {
-		_, err = sheetsClient.Spreadsheets.Values.Append(spreadSheetId, spreadSheetTab+"!A1", &vr).ValueInputOption("RAW").Do()
+	if sheetsClient == nil {
+		return fmt.Errorf("[GoogleSheets] error  :: Failed to initialize google-sheets client")
 	}
+
+	_, err = sheetsClient.Spreadsheets.Values.Append(spreadSheetId, spreadSheetTab+"!A1", &vr).ValueInputOption("RAW").Do()
+
 	return err
 }
 
 // Method to return array of values from a json.
 // source is the json object from transformer and we are iterating the json as a map
 // and we are storing the data into designated position in array based on transformer
 // mappings.
-// Example payload we have from transformer:
+// Example payload we have from transformer without batching:
 // {
 //		message:{
 //			1: { attributeKey: "Product Purchased", attributeValue: "Realme C3" }
 //			2: { attributeKey: "Product Value, attributeValue: "5900"}
 //			..
 // 		}
 // }
-func parseTransformedData(source gjson.Result) ([]string, error) {
-	messagefields := source.Get("message")
-	values := make([]string, len(messagefields.Map()))
-	var pos int
-	var err error
-	if messagefields.IsObject() {
-		for k, v := range messagefields.Map() {
-			pos, err = strconv.Atoi(k)
-			if err != nil {
-				return values, err
+// Example Payload we have from transformer with batching:
+// {
+// 		btach:[
+//			{
+//				message: {
+//					1: { attributeKey: "Product Purchased", attributeValue: "Realme C3" }
+//					2: { attributeKey: "Product Value, attributeValue: "5900"}
+//					..
+// 				}
+//			},
+//			{
+//				message: {
+//					1: { attributeKey: "Product Purchased", attributeValue: "Realme C3" }
+//					2: { attributeKey: "Product Value, attributeValue: "5900"}
+//					..
+// 				}
+//			}
+// 		]
+// }
+func parseTransformedData(source gjson.Result) ([][]interface{}, error) {
```

Comment:
Is it possible to write a few tests for this function only?

Few reasons:
1. Ensure backwards compatibility is supported (we should support it for a long time)
2. Parsing `gjson` is not straightforward, I would like some tests to increase our confident in what ti does and doesn't

Reply:
[Table tests](https://dave.cheney.net/2019/05/07/prefer-table-driven-tests) would be a good way to test this

Reply:
Sure I'll address this

Reply:
@lvrach Can we do it in next sprint for now I have manually tested it with batching and without batching along with backward compatibility check.